src/source_map.rs

```diff
@@ -0,0 +1,125 @@
+// Copyright 2018-2022 the Deno authors. All rights reserved. MIT license.
+
+use crate::module_specifier::ModuleSpecifier;
+
+use deno_ast::ParsedSource;
+use lazy_static::lazy_static;
+use regex::Regex;
+use serde_json::from_slice;
+use serde_json::Value;
+use sourcemap::decode_data_url;
+use sourcemap::DecodedMap;
+
+lazy_static! {
+  static ref SOURCE_MAP_URL_RE: Regex =
+    Regex::new(r#"(?i)^# sourceMappingURL=([^\s'"]+)"#).unwrap();
+}
+
+#[derive(Debug, PartialEq, Eq)]
+pub(crate) enum ParsedSourceMap {
+  Value(Value),
+  Url(ModuleSpecifier),
+}
+
+fn find_source_map_url(parsed_source: &ParsedSource) -> Option<String> {
+  for comment in parsed_source.comments().trailing_map().values() {
```

Comment:
I think this will be in an unstable order so if there are multiple of these source map url comments it may match one or the other. If we're only looking for the trailing comments in a file, then think this would work: `parsed_source.comments().get_trailing(parsed_source.program.span().hi)` (but I forget how swc stores the trailing comments... maybe it's on the last token?)